src/dotnet/MsbuildProject.cs

```diff
@@ -16,44 +19,49 @@ internal class MsbuildProject
     {
         const string ProjectItemElementType = "ProjectReference";
 
-        public ProjectRootElement Project { get; private set; }
+        public ProjectRootElement ProjectRoot { get; private set; }
         public string ProjectDirectory { get; private set; }
 
-        private MsbuildProject(ProjectRootElement project)
+        private ProjectCollection _collection;
+        private List<NuGetFramework> _cachedTfms = null;
+        private Project _cachedEvaluatedProject = null;
+
+        private MsbuildProject(ProjectCollection collection, ProjectRootElement project)
```

Comment:
@rainersigwald's thoughts will be appreciated. CLI should be a fairly exemplary use case of the MSBuild APIs as folks will know to look here for patterns they should use in their own apps.